doc/SONiC_QoS_Mgmt_HLD-1.0.md

```diff
@@ -0,0 +1,499 @@
+
+This data model lacks certain definitions for SONIC QoS feature. Some features that the data model supports also has limitations for future feature expansion. 
+
+As a result, our data model will be loosely based on Open Config YANG, with our enhancement and  modifications on top of it.
+Here is a list of proposed new data model or existing data models from Open Config.
+
+- WRED profile  
+    No Open Config YANG available.  
+    Augment Open Config YANG model with new definitions similar to [SONIC YANG for WRED profile](https://github.com/project-arlo/sonic-mgmt-framework/blob/transformer-phase1/src/cvl/testdata/schema/sonic-wred-profile.yang)
+  
+
+- Scheduler Policy  
+	openconfig-qos-elements.yang::qos-scheduler-policy-config
+     
+
+- DSCP to TC map  
+    No Open Config YANG available.  
+    Augment Open Config YANG model with new definitions similar to [SONIC YANG for DSCP-to-TC-map](https://github.com/project-arlo/sonic-mgmt-framework/blob/transformer-phase1/src/cvl/testdata/schema/sonic-dscp-tc-map.yang)
+  
+
+- TC to Queue map  
+     No Open Config YANG available.   
+     Augment Open Config YANG model with new definition similar to [SONIC YANG for TC-to-Queue-map](https://github.com/project-arlo/sonic-mgmt-framework/blob/transformer-phase1/src/cvl/testdata/schema/sonic-tc-queue-map.yang)
+  
+
+- Interface QoS  
+	openconfig-qos-interfaces.yang:: qos-interfaces-config
+	openconfig-qos-elements.yang:: qos-queue-config
+
+  
+
+- QoS Statistics  
+    openconfig-qos-interfaces.yang:: qos-interfaces/interface/interface-id/output/queues/queue/state  
+    Augment Open Config YANG model with new definitions for queue statistics and watermark statistics.
+
+     
+- Threshold  
+    Augment Open Config YANG model with new definitions of Queue Thresholds.
+
+
+
+### 3.2.2 CLI
+
+#### 3.2.2.1 Configuration Commands
+
+- Configuring WRED
+````
+sonic(config)# wred <name>
+sonic(conf-wred-<name>)# random-detect color [green|yellow|red] minimum-threshold 100 maximum-threshold 300
+drop-probability 40
+sonic(conf-wred-<name>)# random-detect ecn
+
+    sonic(config)# no wred <name>
+````
+
+- Configure Scheduler Policy
+````
+  sonic(config)# scheduler-policy <name>
+  sonic(conf-sched-policy)# scheduler <q#> 
+  sonic(conf-sched-policy-q)# priority level <val>
+  sonic(conf-sched-policy-q)# shape-bandwidth min <val> min-burst <val> max <val> max-burst <val> 
+  sonic(conf-sched-policy-q)# bandwidth-percent <val>
+
+  sonic(conf-sched-policy)# no scheduler <q#>
+  sonic(config)# no scheduler-policy <name>   
+````
+
+
+- Config DSCP to Traffic Class map  
+````
+  sonic(config)# qos-map dscp-to-tc-map <name>
+  sonic(conf-qos-map)# traffic-class <1..8> dscp  {<dscp-value>}
+
+  sonic(config)# no dscp-to-tc-map <name>
+````
+
+- Config Traffic Class to Queue map
+````
+  sonic(config)# qos-map tc-to-queue-map <name>
+  sonic(conf-qos-map)# queue 3 traffic-class 0-3

```

Comment:
since it is traffic class to queue mapping, should be command be "traffic-class 0-3 queue 3".
Also what is the default mapping of traffic-class to queue.